library/core/src/num/f128.rs

```diff
@@ -0,0 +1,1198 @@
+//! Constants for the `f128` double-precision floating point type.
+//!
+//! *[See also the `f128` primitive type][f128].*
+//!
+//! Mathematically significant numbers are provided in the `consts` sub-module.
+//!
+//! For the constants defined directly in this module
+//! (as distinct from those defined in the `consts` sub-module),
+//! new code should instead use the associated constants
+//! defined directly on the `f128` type.
+
+#![unstable(feature = "f128", issue = "none")]
+
+use crate::convert::FloatToInt;
+#[cfg(not(test))]
+use crate::intrinsics;
+use crate::mem;
+use crate::num::FpCategory;
+
+/// Basic mathematical constants.
+#[unstable(feature = "f128", issue = "none")]
+pub mod consts {
+    // FIXME: replace with mathematical constants from cmath.
+
+    /// Archimedes' constant (π)
+    #[unstable(feature = "f128", issue = "none")]
+    pub const PI: f128 = 3.14159265358979323846264338327950288_f128;
```

Comment:
Should we care about implementation of float literal parser? And need to check emitted form of such constants in LLVM IR..

Reply:
I haven’t updated any of the consts but need to do so. Any chance you know of a table anywhere?

Tough to get them from C when you can’t print f128

Reply:
We can easily get rather precise result using arbitary computation tool, it seems here the issue is (1) the smallest digit of such type can tell difference; (2) how compiler parses float literals.

I tried Clang. Disappointingly, it can't differentiate `3.14159265358979q` (or `3.14159265358979L`) with more digits (in generated LLVM IR). So I'm guessing if rustc has similar issue. Maybe analyzing (1) only become reasonable after confirming/fixing (2).

Reply:
I think the parsing comes from `rustc_apfloat` which is a port of LLVM's implementation, so I'd expect it to be about the same.

I think this is the relevant part: https://doc.rust-lang.org/stable/nightly-rustc/rustc_apfloat/ieee/struct.IeeeFloat.html#method.from_decimal_string

Reply:
If you know how to use the tools, would you be able to check that these constants have enough precision for f128? I just used the same values that f64 does and I think it's enough, but not yet able to check

Reply:
I think that I have this covered for now so I will mark this resolved. Please feel free to review the implementation now that it is done @ecnelises